Directory.Packages.props

```diff
@@ -29,6 +29,7 @@
 
     <PackageVersion Include="Microsoft.Extensions.Configuration" Version="8.0.0" />
     <PackageVersion Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="8.0.0" />
+    <PackageVersion Include="Microsoft.Extensions.Diagnostics" Version="8.0.0" />
```

Comment:
Could we use `Microsoft.Extensions.Diagnostics.Abstractions` instead? What other new API are we using other than `IMetricsListener`?

Reply:
I pondered this too. Pulled down the code and tried it out. Not possible because of the use of `IMetricsBuilder.AddMetrics()` which is not in the abstractions package.

Reply:
I doubt you need to call AddMetrics() though. It looks like the call to AddMetrics() is just to get the IMetricsBuilder so that you can call UseOpenTelemetry() on it. Instead of invoking UseOpenTelemetry() you could instead call:
```C#
var builder = new MeterProviderBuilderBase(services);
services.TryAddEnumerable(
       ServiceDescriptor.Singleton<IMetricsListener,OpenTelemetryMetricsListener>());
configure(builder);
```

Reply:
> It looks like the call to AddMetrics() is just to get the IMetricsBuilder so that you can call UseOpenTelemetry() on it.

Not exactly. I went back and forth on this. Decided to include the call to `AddMetrics` so that the underlying `IServiceCollection` would always have a consistent worldview. If it just dropped in an `IMetricsListener` something like this may be surprising...

```csharp
var services = new ServiceCollection();
services.AddOpenTelemetry().WithMetrics();

using var sp = services.BuildServiceProvider();

var meterFactory = sp.GetRequiredService<IMeterFactory>(); // Will throw without AddMetrics
```

That is kind of half the story. To make sure all the required services are there.

The other half of the story...

The goal is also for `WithLogging` to do the same work as `appBuilder.Logging.AddOpenTelemetry()`. `WithLogging` is more complex. A simplified version for it looks like...

```csharp
services
   .AddLogging();
   .AddConfiguration(); // Required for options binding and needs a logging builder to do that

// Also required for options binding
LoggerProviderOptions.RegisterProviderOptions<OpenTelemetryLoggerOptions, OpenTelemetryLoggerProvider>(services);

// The actual provider registration
services.TryAddEnumerable(ServiceDescriptor.Singleton<ILoggerProvider, OpenTelemetryLoggerProvider);
```

I thought since "WithLogging" will need to call "services.AddLogging()" that "WithMetrics" should also call "services.AddMetrics()".

I'm open to changing this, just wanted to provide more context.

Reply:
I just lowered the reference from `Microsoft.Extensions.Diagnostics` to `Microsoft.Extensions.Diagnostics.Abstractions`. I don't feel confident that this is the totally correct thing to do, but I figured we could always raise it again if needed.